taliforth.asm

```diff
@@ -227,6 +227,20 @@ dovar:
 ; =====================================================================
 ; LOW LEVEL HELPER FUNCTIONS
 
+push_upvar_tos:
```

Comment:
I probably should have done this by now.  Just so you are aware, Tali already has a non-standard word `useraddr` which gives the address (in zero page) of `UP`.  Using that, it's possible to pull the base address of the user page varibles, add an offset, and dereference to get access to these values from Forth (does require knowledge of the offsets, of course).

Reply:
oh, interesting.  it does seem at some point that collapsing out `UP` and moving to a "single user" implementation would offer some space/time/complexity saving.   Currently `UP` never changes right?

Reply:
That is correct - currently UP never changes once set by `quit`